src/cmap/connection_pool.ts

```diff
@@ -99,12 +107,18 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
    * @internal
    */
   [kGeneration]: number;
+  /** A map of generations to service ids
+   * @internal
+   */
+  [kServiceGenerations]: Map<ObjectId, number>;
```

Comment:
The generation in LB mode is a map of `serviceId` to generation number.

Reply:
@durran Is it intentional that this is a `Map<ObjectId, ...>`? I think this might end up being surprising when working the code here considering that Maps work based on object identity, not equality – maybe a `Map<string, number>` would be better?

Reply:
Yes I think I can change this to the hex string representation of the object id in this case.

Reply:
This is in now.